src/main/java/tool/arenasizer/ArenaInfoPanel.java

```diff
@@ -17,125 +17,148 @@
-
-        expandedCapacityPanel.setTranslation();
-        expandedCapacityPanel.setTitle(TranslationFacility.tr("ArenaInfoPanel.expansion"));
-        expandedCapacityPanel.labelPercentage.setText(TranslationFacility.tr("ArenaInfoPanel.change_in_percent"));
-        expandedCapacityPanel.label1.setText(TranslationFacility.tr("ArenaInfoPanel.completion"));
-
-        futureCapacityPanel.setTranslation();
-        futureCapacityPanel.setTitle(TranslationFacility.tr("ArenaInfoPanel.future"));
-        futureCapacityPanel.label1.setText(TranslationFacility.tr("ArenaInfoPanel.available"));
-    }
-
-    private void updateExpansionFinishedLabel() {
-        final JLabel labelExpansionFinished = expandedCapacityPanel.label3;
-        final Stadium stadium = HOVerwaltung.instance().getModel().getStadium();
-        Optional.ofNullable(stadium).map(Stadium::getExpansionDate).ifPresentOrElse(expansionDate -> {
-                    final boolean building = expansionDate.isAfter(HODateTime.now());
-                    if (building) {
-                        final String text = toDurationString(expansionDate);
-                        final String toolTipText = String.format(
-                                TranslationFacility.tr("ArenaInfoPanel.finished_in_n_days_format"),
-                                HODateTime.daysFromNow(expansionDate, 1));
-                        labelExpansionFinished.setText(text);
-                        labelExpansionFinished.setToolTipText(toolTipText);
-                    } else {
-                        labelExpansionFinished.setText(String.format(TranslationFacility.tr("ArenaInfoPanel.built")));
-                        labelExpansionFinished.setToolTipText(null);
-                    }
-                },
-                () -> {
-                    labelExpansionFinished.setText(EMPTY);
-                    labelExpansionFinished.setToolTipText(null);
-                }
-        );
-    }
-
-    private static String toDurationString(HODateTime hoDateTime) {
-        return HumanDuration.of(HODateTime.between(HODateTime.now(), hoDateTime)).toHumanString();
-    }
-}
\ No newline at end of file
+	private final CapacityPanel currentCapacityPanel;
+	private final CapacityPanel expandedCapacityPanel;
+	private final CapacityPanel futureCapacityPanel;
+
+	public ArenaInfoPanel() {
+		setLayout(new FlowLayout());
+
+		currentCapacityPanel = new CapacityPanel();
+		add(currentCapacityPanel);
+
+		expandedCapacityPanel = new CapacityPanel();
+		add(expandedCapacityPanel);
+
+		futureCapacityPanel = new CapacityPanel();
+		add(futureCapacityPanel);
+
+		updateValues();
+
+		Timer timer = new Timer(1000, actionEvent -> updateExpansionFinishedLabel());
+		timer.start();
+	}
+
+	private BigDecimal calculatePercentExpansion(int numberOfSeats, int expansion) {
+		if (numberOfSeats == 0) {
+			return null; // Ideally we'd have a way to represent ∞, maybe future update.
+		} else {
```

Comment:
This is the heart of the fix, checking that existing seating is 0.

Reply:
Good fix - alternativly we could refer to the current value. then 100% would be the displayed value in this case.

if ( currentValue > 0 ) percentage = expansion / currentValue
else if (expasion !=0) percentage = -100
else percentage = 0